litex/tools/litex_sim.py

```diff
@@ -62,6 +62,72 @@ class Platform(SimPlatform):
     def __init__(self):
         SimPlatform.__init__(self, "SIM", _io)
 
+# DFI PHY model settings ---------------------------------------------------------------------------
+
+sdr_phy_settings = {
+    "nphases": 1,
+    "rdphase": 0,
+    "wrphase": 0,
+    "rdcmdphase": 0,
+    "wrcmdphase": 0,
+    "cl": 2,
+    "read_latency": 4,
+    "write_latency": 0
+}
+
+sdram_module_rates = {
+    "SDR": "1:1",
+    "DDR": "1:2",
+    "LPDDR": "1:2",
+    "DDR2": "1:2",
+    "DDR3": "1:4",
+}
+
+def get_sdram_phy_settings(memtype, data_width, clk_freq):
+    nphases = 4 if memtype == "DDR3" else 2
+
+    # Settings taken from s7ddrphy
+    if memtype in ["DDR2", "DDR3"]:
+        tck                 = 2/(2*nphases*clk_freq)
```

Comment:
We should probably organize things in the order of the complexity: SDR, then DDR/LPDDR, then DDR2/DDR3.

Reply:
This is in correct order now